src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,112 @@
+                        PreparedStatement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setInt(2, book.getPrice());
+            if (statement.executeUpdate() < 1) {
+                throw new RuntimeException("Should be at least one row affected, but got zero");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot create for book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM book WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            if (resultSet.next()) {
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(resultSet.getObject("title", String.class));
+                book.setPrice(resultSet.getObject("price", Integer.class));
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot find by id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM book";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> bookList = new ArrayList<>();
+            while (resultSet.next()) {
+                Book book = new Book();
+                book.setId(resultSet.getObject("id", Long.class));
+                book.setTitle(resultSet.getObject("title", String.class));
+                book.setPrice(resultSet.getObject("price", Integer.class));
+                bookList.add(book);
+            }
+            return bookList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot find rows", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE book SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            if (book.getId() == null) {
+                throw new RuntimeException("Id cannot be null");
+            }
+            statement.setString(1, book.getTitle());
+            statement.setInt(2, book.getPrice());
```

Comment:
The `setInt` method is used for setting the `price`. Change this to `setBigDecimal` and ensure the `price` is of type `BigDecimal`.